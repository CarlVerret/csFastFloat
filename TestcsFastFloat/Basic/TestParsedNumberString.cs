using csFastFloat.Structures;
using System.Collections.Generic;
using System.Text;
using Xunit;

namespace TestcsFastFloat.Tests.Basic
{
#pragma warning disable xUnit1026



  public class TestParsedNumberString : BaseTestClass
  {
    [Fact]
    unsafe public void ParseNumberString_Works_Scnenarios()
    {
      Dictionary<string, string> sut = new Dictionary<string, string>();

      sut.Add("leading zeros", "001");
      sut.Add("leading zeros neg", "-001");

      sut.Add("zero", "0");
      sut.Add("zero neg", "-0");

      sut.Add("double", "0.00000000000000212312312");
      sut.Add("double neg", "-0.00000000000000212312312");
      sut.Add("int", "1");
      sut.Add("int neg", "-1");

      sut.Add("autreint ", "123124");
      sut.Add("autreint neg", "-123124");

      sut.Add("notation scientifique", "4.56E+2");
      sut.Add("notation scientifique neg", "-4.56E-2");

      sut.Add("notation scientifique 2", "4.5644E+2");
      sut.Add("notation scientifique 2 neg", "-4.5644E-2");

      sut.Add("notation scientifique 3", "4424.5644E+22");
      sut.Add("notation scientifique 3 neg", "-4424.5644E-22");

      sut.Add("notation scientifique 4", "4424.5644E+223");
      sut.Add("notation scientifique 4 neg", "-4424.5644E-223");
      StringBuilder sb = new StringBuilder();

      foreach (KeyValuePair<string, string> kv in sut)
      {
        sb.AppendLine($"Scenario : {kv.Key} ");
        sb.AppendLine($"Valeur   : {kv.Value} ");

        fixed (char* p = kv.Value)
        {
          char* pend = p + kv.Value.Length;
          var res = ParsedNumberString.ParseNumberString(p, pend);

          sb.AppendLine($"Resultat : {res.exponent} {res.mantissa} {res.negative} {res.valid}");
          sb.AppendLine();
        }
      }

      VerifyData(sb.ToString());
    }


  }
}