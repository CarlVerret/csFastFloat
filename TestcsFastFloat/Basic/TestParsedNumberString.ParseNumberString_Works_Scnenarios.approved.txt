Scenario : leading zeros 
Valeur   : 001 
Resultat : 0 1 False True

Scenario : leading zeros neg 
Valeur   : -001 
Resultat : 0 1 True True

Scenario : zero 
Valeur   : 0 
Resultat : 0 0 False True

Scenario : zero neg 
Valeur   : -0 
Resultat : 0 0 True True

Scenario : double 
Valeur   : 0.00000000000000212312312 
Resultat : -23 212312312 False True

Scenario : double neg 
Valeur   : -0.00000000000000212312312 
Resultat : -23 212312312 True True

Scenario : int 
Valeur   : 1 
Resultat : 0 1 False True

Scenario : int neg 
Valeur   : -1 
Resultat : 0 1 True True

Scenario : autreint  
Valeur   : 123124 
Resultat : 0 123124 False True

Scenario : autreint neg 
Valeur   : -123124 
Resultat : 0 123124 True True

Scenario : notation scientifique 
Valeur   : 4.56E+2 
Resultat : 0 456 False True

Scenario : notation scientifique neg 
Valeur   : -4.56E-2 
Resultat : -4 456 True True

Scenario : notation scientifique 2 
Valeur   : 4.5644E+2 
Resultat : -2 45644 False True

Scenario : notation scientifique 2 neg 
Valeur   : -4.5644E-2 
Resultat : -6 45644 True True

Scenario : notation scientifique 3 
Valeur   : 4424.5644E+22 
Resultat : 18 44245644 False True

Scenario : notation scientifique 3 neg 
Valeur   : -4424.5644E-22 
Resultat : -26 44245644 True True

Scenario : notation scientifique 4 
Valeur   : 4424.5644E+223 
Resultat : 219 44245644 False True

Scenario : notation scientifique 4 neg 
Valeur   : -4424.5644E-223 
Resultat : -227 44245644 True True

